parser-typechecker/src/Unison/Codebase/FileCodebase.hs

```diff
@@ -123,20 +127,26 @@ initCodebaseAndExit mdir = do
   _ <- initCodebase dir
   exitSuccess
 
+absDir :: FilePath -> IO FilePath
+absDir dir 
+  | dir == "." = makeAbsolute dir
+  | dir == ".." = takeDirectory <$> makeAbsolute "."
```

Comment:
WDYT about using `canonicalizePath` instead of `takeDirectory` if that’s what you want to do; in fact I guess `absDir` is really just trying to implement `canonicalizePath`, right?

Reply:
Good catch; I am not as familiar as I would like to be with Haskell libraries, so in this instance I read the bold note and  shied away from using `canonicalizePath`. (...and ended up reimplementing it naively of course.) 

Will update this :+1: .

Reply:
It’s okay, we’re all learning.